### PR TITLE
SRCH-1822 update code and templates for compatibility with Elasticsearch 7

### PR DESCRIPTION
**Pull request by MothOnMars**
This PR encompasses all the changes necessary for compatibility with Elasticsearch 7. Changes include:
- adding ES/Kibana 7 to our local Docker setup 
- upgrading the Elasticsearch gems
- implement repository pattern
- replace `patron` with `typhoeus`
- running specs against ES 7

### CODE REVIEW

**Comment by MothOnMars** [comment]
@jmax-fearless , this is ready for review. This is the release branch that encompasses all the ES 7 changes that have been reviewed in past PRs. Review at this point should cover the basic sanity-checks:
- [ ] are there any outstanding `TODO`'s or `FIXME`s?
- [ ] are there any oopsies, like editor files that should not be committed, or stray commits that don't belong in this branch?

File: Gemfile.lock [hunk 1]
```diff
@@ -102,32 +109,35 @@ GEM
...
-      elasticsearch-api (= 5.0.4)
-      elasticsearch-transport (= 5.0.4)
-    elasticsearch-api (5.0.4)
+    elasticsearch (6.8.2)
+      elasticsearch-api (= 6.8.2)
+      elasticsearch-transport (= 6.8.2)
+    elasticsearch-api (6.8.2)
       multi_json
-    elasticsearch-dsl (0.1.5)
-    elasticsearch-model (5.0.2)
```
Comment (jmax-fearless) [inline Gemfile.lock:111]: That's quite a jump.

**Review by jmax-fearless — APPROVED** [review]
Approved.

**Comment by MothOnMars** [comment]
Must be deployed with https://github.com/GSA/usasearch-cookbooks/pull/477

**Review by dawnpm — APPROVED** [review]

**Pull request merged**